walkdir.py

```diff
@@ -276,10 +287,21 @@ def file_paths(walk_iter):
 
 def all_paths(walk_iter):
     """Iterate over both files and directories visited by the underlying walk"""
-    for dirpath, subdirs, files in walk_iter:
-        yield dirpath
-        for fname in files:
+    dir_entry = next(walk_iter, None)
+    if dir_entry is None:
+        return
+    top = dir_entry[0]
+    yield top
+    while dir_entry:
+        dirpath = dir_entry[0]
+        if top not in dirpath:
```

Comment:
Ah, now I properly understand your comment on the PR - when "min_depth" is set, it *is* like having multiple independent walks, so there's an output difference where those independent roots now appear immediately before their contents, rather than all together as part of the root directory's listing.

I think that's fine - it just needs to be documented appropriately. On `min_depth` itself, that would be a note explaining the output is like calling `itertools.chain` on multiple underlying walks, while the docs for `dir_paths` and `all_paths` need to cover how walks with multiple independent roots are handled, as well as a versionchanged note explaining the switch in behaviour between 0.3 and 0.4.